### PR TITLE
Set cell color by selected attribute

### PR DESCRIPTION
**Pull request by jrsconfitto**
This allows the color of cells to be influenced by the value of the selected (EF?) attribute.

### CODE REVIEW

**Comment by jrsconfitto** [comment]
Still needs some work, but i have something to show! Here's a treemap tweaking colors by the Duration attribute:

image

i'm also logging colors to the `debug` console, which is a nice little trick to compare what the color would be under our old model to what's now being shown:

image

**Comment by jrsconfitto** [comment]
Alright! Coloring by two different attribute values now:

image

**Comment by jrsconfitto** [comment]
i'll test on the test system and then merge if it works!

**Comment by jrsconfitto** [comment]
OK, need to create the template selector on the symbol's config page.

**Comment by jrsconfitto** [comment]
OK, i may try to see if there's a way to get the colors more along a spectrum? Not sure how worth our time that is at this point, though. That can be tackled later... merging!

**Comment by FishmanD** [comment]
I really like this a lot.  I will see if I can add a dummy operator2 in since the bad OSIsoft provided has the same operator.  There might be a better data set we could use on the server.  I added in operator2 as either 0 or 1 and it looks pretty sweet.  I noticed our PI api calls don't work well with enumerations sets, which is not a big deal.

File: app.js [hunk 1]
```diff
@@ -295,15 +327,15 @@ var eventsModule = function () {
             var bulkQuery = {};
             templateUsed.frames.forEach(EF => {
                 var attributeURL;
-                attributeURL = encodeURI(webAPIServerURL + "/streamsets/" + EF.id + "/value?nameFilter=" + attributeName + "&selectedFields=Items.Value.Value");
+                attributeURL = encodeURI(webAPIServerURL + "/streamsets/" + EF.id + "/value?selectedFields=Items.Name;Items.Value.Value");
```
Comment (FishmanD) [inline app.js:330]: I think it makes sense to get all of the attributes now that we have a use!!  Not much of a performance cost due to how the web api does the caching for EF attributes.

File: app.js [hunk 2]
```diff
@@ -228,9 +260,9 @@ var eventsModule = function () {
         GetAllTemplateAttributes();
 
         // Get attribute value for provide attribute and template.
-        if (_attribute != undefined && _attribute != "None" && _template != "None") {
+        if (_sizeAttribute != undefined && _sizeAttribute != "None" && _template != "None") {
```
Comment (FishmanD) [inline app.js:263]: Small typo perhaos with _sizeAttribute maybe...

Comment (jrsconfitto) [reply app.js:263]: Sorry! i'm not seeing the typo 🙈 , where is it?

**Comment by jrsconfitto** [comment]
> There might be a better data set we could use on the server. I added in operator2 as either 0 or 1 and it looks pretty sweet. I noticed our PI api calls don't work well with enumerations sets, which is not a big deal.

i was thinking that we could now start looking for good data to show and wondered about categorizing by enumerations too. i think we're on the same page here.

**Pull request merged**